app/controllers/util/DynamicFormUtils.scala

```diff
@@ -0,0 +1,259 @@
+            val label: String = if (inputInfo.length == 3) inputInfo(2) else enteredValueAsString.capitalize
+            (label, enteredValueAsString)
+        }.toSeq
+
+        FieldValues(
+          fieldId = metadataName,
+          defaultFieldValue.fieldOptions,
+          selectedFieldOption = Some(labelAndEnteredValue),
+          defaultFieldValue.multiValueSelect,
+          defaultFieldValue.fieldLabel,
+          defaultFieldValue.fieldHint,
+          defaultFieldValue.fieldRequired,
+          fieldError = valuesBelongingToMetadataName.find { case (_, (_, errors)) => errors.nonEmpty } match {
+            case Some((_, (_, errors))) => errors
+            case None => defaultFieldValue.fieldError // the default in an empty list
+          }
+        ) ->
+          dataType
+    }
+  }
+
+  def validateFormAnswers(formAnswers: Map[String, Seq[String]]): Map[String, (Option[Any], List[String])] = {
+    val validatedFields: Map[String, (Option[Any], List[String])] = formAnswers.map {
+      case (inputName, valueSelected) => validateFields(inputName, valueSelected.head)
+    }
+    val dateFieldsWhereAllUnitsOfTimeAreCorrect: Map[String, Map[String, (Option[Any], List[String])]] = getFieldsWhereDMYAreAllCorrect(validatedFields)
+
+    val numberOfDaysInMonthChecks: Map[String, (Option[Any], List[String])] =
+      dateFieldsWhereAllUnitsOfTimeAreCorrect.map {
+        case (_, dateFieldWhereAllUnitsOfTimeAreCorrect) =>
+          val fieldsThatHaveHadDaysValidated: (String, (Option[Any], List[String])) = validateDaysInMonth(dateFieldWhereAllUnitsOfTimeAreCorrect)
+          fieldsThatHaveHadDaysValidated
+      }
+    val fullyValidatedFields: Map[String, (Option[Any], List[String])] = validatedFields ++ numberOfDaysInMonthChecks
+    fullyValidatedFields
+  }
+
+  private def validateFields(inputName: String, valueSelected: String): (String, (Option[Any], List[String])) = {
+    val inputInfo = inputName.split("-")
+    val fieldType = inputInfo.head
+    val metadataName = inputInfo(1)
+    checkThatMetadataNameIsExists(inputName, fieldType, metadataName, inputInfo, valueSelected)
+  }
+
+  private def checkThatMetadataNameIsExists(inputName: String, fieldType: String, metadataName: String,
+                                            inputInfo: Array[String], valueSelected: String): (String, (Option[Any], List[String])) = {
+    val fieldValueBelongingToMetadataName = defaultFieldValues.find {
+      case (fieldValues, _) => fieldValues.fieldId == metadataName
+    }
+
+    fieldValueBelongingToMetadataName match {
+      case Some((fieldValue, _)) =>
+        validateFieldType(inputName, fieldType, metadataName, inputInfo, valueSelected, fieldValue)
+      case None => throw new IllegalArgumentException(s"Metadata name $metadataName, extracted from field input name, does not exist.")
+    }
+  }
+
+  private def validateFieldType(inputName: String, fieldType: String, metadataName: String,
```

Comment:
There are text fields in the database, are they covered in this method?

Reply:
No, they aren't. I haven't made any checks for input text yet since we do not have a `text` input box on any pages yet (just numeric and date); I was planning on adding it/letting someone else add it when I/they get to it as it's a little more involved and requires more thought (UTF-8 checking, escaping characters?). For now, it will throw an exception forcing someone to create an implementation for it. Should I add  a check for plain text now?

Reply:
I think that's best. Just put something in that allows any text. We can worry about input validation later once we know what to do. It would be good to get this in a state where we can put any valid entries in the database and it will work.